### PR TITLE
Naming screen

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/naming_screen.c [hunk 1]
```diff
@@ -0,0 +1,2400 @@
...
+static void sub_809E6B8(u8 a0);
+static void sub_809E6E0(struct Task * task, u8 a1, u8 a2);
+static void sub_809E700(struct Sprite * sprite);
+static void sub_809E7F0(struct Sprite * sprite);
+static void sub_809E83C(struct Sprite * sprite);
+static void sub_809E898(void);
+static void CursorInit(void);
+static void SetCursorPos(s16 x, s16 y);
+static void GetCursorPos(s16 *xP, s16 *yP);
+static void MoveCursorToOKButton();
```
Comment (jiangzhengwenjz) [inline src/naming_screen.c]: missing `void`

File: src/naming_screen.c [hunk 2]
```diff
@@ -0,0 +1,2400 @@
...
+static void sub_809EA0C(u8 a0);
+static void sub_809EA64(u8 a0);
+static bool8 IsCursorAnimFinished();
```
Comment (jiangzhengwenjz) [inline src/naming_screen.c]: missing `void`

File: src/naming_screen.c [hunk 3]
```diff
@@ -0,0 +1,2400 @@
...
+    task->tKeyboardEvent = 0;
+
+    if (gMain.newKeys & A_BUTTON)
+        task->tKeyboardEvent = KBEVENT_PRESSED_A;
+    else if (gMain.newKeys & B_BUTTON)
+        task->tKeyboardEvent = KBEVENT_PRESSED_B;
+    else if (gMain.newKeys & SELECT_BUTTON)
+        task->tKeyboardEvent = KBEVENT_PRESSED_SELECT;
+    else if (gMain.newKeys & START_BUTTON)
+        task->tKeyboardEvent = KBEVENT_PRESSED_START;
```
Comment (jiangzhengwenjz) [inline src/naming_screen.c]: `JOY_NEW`

File: src/naming_screen.c [hunk 4]
```diff
@@ -0,0 +1,2400 @@
...
+    GetCursorPos(&cursorX, &cursorY);
+    dpadDir = 0;
+    if (gMain.newAndRepeatedKeys & DPAD_UP)
+        dpadDir = 1;
+    if (gMain.newAndRepeatedKeys & DPAD_DOWN)
+        dpadDir = 2;
+    if (gMain.newAndRepeatedKeys & DPAD_LEFT)
+        dpadDir = 3;
+    if (gMain.newAndRepeatedKeys & DPAD_RIGHT)
+        dpadDir = 4;
```
Comment (jiangzhengwenjz) [inline src/naming_screen.c]: `JOY_REPT`

File: src/naming_screen.c [hunk 5]
```diff
@@ -0,0 +1,2400 @@
...
+
+static const u8 sGenderColors[2][3] = {
+    {TEXT_COLOR_TRANSPARENT, TEXT_COLOR_LIGHT_BLUE, TEXT_COLOR_BLUE},
+    {TEXT_COLOR_TRANSPARENT, TEXT_COLOR_LIGHT_RED, TEXT_COLOR_RED}
+};
+
+static void AddGenderIconFunc_Yes(void)
+{
+    u8 genderSymbol[2];
+    bool8 isFemale = FALSE;
```
Comment (jiangzhengwenjz) [inline src/naming_screen.c]: I think it's just a gender variable which uses `MALE` and `FEMALE`

File: src/naming_screen.c [hunk 6]
```diff
@@ -0,0 +1,2400 @@
...
+
+static const struct NamingScreenTemplate rivalNamingScreenTemplate = {
+    .copyExistingString = FALSE,
+    .maxChars = OT_NAME_LENGTH,
+    .iconFunction = 4,
+    .addGenderIcon = 0,
+    .initialPage = KBPAGE_LETTERS_UPPER,
+    .title = gText_RivalsName,
+};
+
```
Comment (jiangzhengwenjz) [inline src/naming_screen.c]: the names of above templates should start with `s`

File: src/naming_screen.c [hunk 7]
```diff
@@ -0,0 +1,2400 @@
...
+    },
+    [KBPAGE_SYMBOLS] = {
+        gText_NamingScreenKeyboard_01234,
+        gText_NamingScreenKeyboard_56789,
+        gText_NamingScreenKeyboard_Symbols1,
+        gText_NamingScreenKeyboard_Symbols2
+    },
+};
+
+// FIXME: Sync with Emerald
```
Comment (jiangzhengwenjz) [inline src/naming_screen.c]: What do you mean?

File: src/naming_screen.c [hunk 8]
```diff
@@ -0,0 +1,2400 @@
...
+static const struct SpriteSheet gUnknown_83E267C[] = {
+    {gUnknown_8E98858, 0x1E0,  0x0000},
+    {gUnknown_8E98A38, 0x1E0,  0x0001},
+    {gUnknown_8E985D8, 0x280,  0x0002},
+    {gUnknown_8E98FD8, 0x100,  0x0003},
+
```
Comment (jiangzhengwenjz) [inline src/naming_screen.c]: {}

Comment (jiangzhengwenjz) [inline src/naming_screen.c]: {}

File: src/naming_screen.c (hunk 7 above)
Comment (PikalaxALT) [reply src/naming_screen.c]: the gfx need to be reorganized to sync up with emerald

**Pull request merged**